RallyHereDebugTool/Source/Private/RH_ImGuiUtilities.cpp

```diff
@@ -166,6 +166,162 @@ void ImGuiDisplayCustomData(const TMap<FString, FString>& CustomData, const FStr
+	{
+		ImGui::TableNextRow();
+		ImGui::TableNextColumn();
+		ImGui::Text(TCHAR_TO_UTF8(*Key));
+		ImGui::TableNextColumn();
+
+		if (IsSetProperty != nullptr)
+		{
+			if (FBoolProperty const* BoolProp = CastField<FBoolProperty>(IsSetProperty))
+			{
+				if (!BoolProp->GetPropertyValue(Data))
+				{
+					ImGui::Text("<UNSET>");
+					return;
+				}
+			}
+		}
+
+		if (FBoolProperty const* BoolProp = CastField<FBoolProperty>(Property))
+		{
+			ImGui::Text(BoolProp->GetPropertyValue(Data) ? "TRUE" : "FALSE");
+		}
+		else if (FNumericProperty const* NumericProperty = CastField<FNumericProperty>(Property))
+		{
+			if (NumericProperty->IsInteger())
+			{
+				ImGui::Text("%s", TCHAR_TO_UTF8(*FString::Printf(TEXT("%d"), NumericProperty->GetSignedIntPropertyValue(Data))));
+			}
+			else
+			{
+				ImGui::Text("%s", TCHAR_TO_UTF8(*FString::Printf(TEXT("%d"), NumericProperty->GetFloatingPointPropertyValue(Data))));
+			}
+		}
+		else if (FEnumProperty const* EnumProperty = CastField<FEnumProperty>(Property))
+		{
+			if (Property->HasMetaData("SerializeAsInteger"))
+			{
+				FNumericProperty* UnderlyingProperty = EnumProperty->GetUnderlyingProperty();
+
+				ImGui::Text("%s", TCHAR_TO_UTF8(*FString::Printf(TEXT("%d"), UnderlyingProperty->GetSignedIntPropertyValue(Data))));
+			}
+			else
+			{
+				FString	ValueStr;
+				Property->ExportText_Direct(ValueStr, Data, nullptr, nullptr, PPF_None);
+
+				ImGui::Text("%s", TCHAR_TO_UTF8(*ValueStr));
+			}
+		}
+		else if (FStrProperty const* StringProperty = CastField<FStrProperty>(Property))
+		{
+			FString	ValueStr;
+			Property->ExportText_Direct(ValueStr, Data, nullptr, nullptr, PPF_None);
+
+			ImGui::Text("%s", TCHAR_TO_UTF8(*ValueStr));
+		}
+		else if (FMapProperty const* MapProp = CastField<FMapProperty>(Property))
+		{
+			if (ImGui::TreeNodeEx(TCHAR_TO_UTF8(*FString::Printf(TEXT("Data##%s"), *Key)), RH_DefaultTreeFlags))
+			{
+				if (ImGui::BeginTable("DataTable", 2, RH_TableFlagsPropSizing))
+				{
+					ImGui::TableSetupColumn("Key");
+					ImGui::TableSetupColumn("Value");
+					ImGui::TableHeadersRow();
+
+					FScriptMapHelper MapHelper(MapProp, Property->ContainerPtrToValuePtr<uint8>(Model));
+					for (int32 i = 0; i < MapHelper.Num(); ++i)
+					{
+						FString	KeyStr;
+						FString	ValueStr;
+						MapHelper.GetKeyProperty()->ExportText_Direct(KeyStr, MapHelper.GetKeyPtr(i), MapHelper.GetKeyPtr(i), nullptr, PPF_None);
+						MapHelper.GetValueProperty()->ExportText_Direct(ValueStr, MapHelper.GetValuePtr(i), MapHelper.GetValuePtr(i), nullptr, PPF_None);
+
+						ImGui::TableNextRow();
+						ImGui::TableNextColumn();
+						ImGui::Text(TCHAR_TO_UTF8(*KeyStr));
+						ImGui::TableNextColumn();
+						ImGui::Text(TCHAR_TO_UTF8(*ValueStr));
+					}
+
+					ImGui::EndTable();
+
+				}
+				ImGui::TreePop();
+			}
+		}
+		else if (FStructProperty const* StructProp = CastField<FStructProperty>(Property))
+		{
```

Comment:
Would add a specific handler for guid properties to print them in our standard method.